certbot-ci/certbot_integration_tests/utils/misc.py

```diff
@@ -241,13 +239,19 @@ def generate_csr(domains, key_path, csr_path, key_type=RSA_KEY_TYPE):
     else:
         raise ValueError('Invalid key type: {0}'.format(key_type))
 
-    key_bytes = crypto.dump_privatekey(crypto.FILETYPE_PEM, key)
-    with open(key_path, 'wb') as file:
-        file.write(key_bytes)
+    with open(key_path, 'wb') as file_h:
+        file_h.write(crypto.dump_privatekey(crypto.FILETYPE_PEM, key))
+
+    req = crypto.X509Req()
+    san = ', '.join(['DNS:{0}'.format(item) for item in domains])
+    san_constraint = crypto.X509Extension(b'subjectAltName', False, san.encode('utf-8'))
+    req.add_extensions([san_constraint])
+
+    req.set_pubkey(key)
+    req.sign(key, 'sha256')
```

Comment:
I think we probably also want to set the version like is done in `make_csr` according to https://www.rfc-editor.org/rfc/rfc2459.html#section-4.1.2.1.

Reply:
Are you sure you want this level of finition here ? I mean, the test is about checking we are able to process a CSR. Do we really mean to have a CSR "by the book"?

Reply:
Yes I think it's worth it. It's a simple change to make and while I think it's very unlikely, without it we could have trouble with the CSR being parsed incorrectly in the future.

Reply:
Oh, it is just about adding the version. OK, doing that.

Reply:
Done.